app/controllers/viewandamend/ViewAmendSelectorController.scala

```diff
@@ -0,0 +1,120 @@
+
+import config.FrontendAppConfig
+import controllers.actions._
+import controllers.viewandamend.routes
+import models.QtStatus.AmendInProgress
+import models.authentication.{PsaUser, PspUser}
+import models.{PstrNumber, QtStatus, SessionData, TransferId}
+import pages.memberDetails.MemberNamePage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import services.UserAnswersService
+import uk.gov.hmrc.mongo.lock.LockRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.viewandamend.ViewAmendSelectorView
+
+import javax.inject.Inject
+import scala.concurrent.duration.DurationLong
+import scala.concurrent.{ExecutionContext, Future}
+
+class ViewAmendSelectorController @Inject() (
+    override val messagesApi: MessagesApi,
+    identify: IdentifierAction,
+    getData: DataRetrievalAction,
+    schemeData: SchemeDataAction,
+    val controllerComponents: MessagesControllerComponents,
+    view: ViewAmendSelectorView,
+    appConfig: FrontendAppConfig,
+    lockRepository: LockRepository,
+    userAnswersService: UserAnswersService,
+    sessionRepository: SessionRepository
+  )(implicit ec: ExecutionContext
+  ) extends FrontendBaseController with I18nSupport {
+
+  private val lockTtlSeconds: Long = appConfig.dashboardLockTtl
+
+  def onPageLoad(qtReference: TransferId, pstr: PstrNumber, qtStatus: QtStatus, versionNumber: String): Action[AnyContent] =
+    (identify andThen schemeData) { implicit request =>
+      Ok(view(qtReference, pstr, qtStatus, versionNumber)).withSession(
+        request.session +
+          ("qtReference"   -> qtReference.value) +
+          ("pstr"          -> pstr.value) +
+          ("qtStatus"      -> qtStatus.toString) +
+          ("versionNumber" -> versionNumber)
+      )
+    }
+
+  def onSubmit(qtReference: TransferId, pstr: PstrNumber, qtStatus: QtStatus, versionNumber: String): Action[AnyContent] =
+    (identify andThen schemeData).async { implicit request =>
+      for {
+        formData <- Future.successful(
+                      request.body.asFormUrlEncoded
+                        .flatMap(_.get("option").flatMap(_.headOption))
+                    )
+        result   <- formData match {
+
+                      case Some("view") =>
+                        Future.successful(Redirect(routes.ViewAmendSubmittedController.view(qtReference, pstr, qtStatus, versionNumber)))
+
+                      case Some("") | None =>
+                        Future.successful(
+                          Redirect(routes.ViewAmendSelectorController.onPageLoad(qtReference, pstr, qtStatus, versionNumber))
+                            .flashing("error" -> "true")
+                        )
+
+                      case Some("amend") =>
+                        val owner = request.authenticatedUser match {
```

Comment:
We should consider refactoring the locking logic into a service or something, it would save getting the authenticated user etc every time. Not necessarily in this ticket though